Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Utilities.Async;
+using Microsoft.MixedReality.Toolkit.Core.Utilities.Async.AwaitYieldInstructions;
+using Microsoft.MixedReality.Toolkit.Utilities.Gltf.Schema;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Gltf.Serialization
+{
+    public static class GltfUtility
+    {
+        public const uint GltfMagicNumber = 0x46546C67;
+
+        private static readonly WaitForUpdate Update = new WaitForUpdate();
+        private static readonly WaitForBackgroundThread BackgroundThread = new WaitForBackgroundThread();
+
+        /// <summary>
+        /// Imports a glTF object from the provided uri
+        /// </summary>
+        /// <param name="uri">the path to the file to load</param>
+        /// <returns>New <see cref="GltfObject"/> imported from uri.</returns>
+        /// <remarks>
+        /// If the <see cref="Application.isPlaying"/> is false, then this method will run synchronously.
+        /// </remarks>
+        public static async Task<GltfObject> ImportGltfObjectFromPathAsync(string uri)
+        {
+            if (string.IsNullOrWhiteSpace(uri))
+            {
+                Debug.LogError("Uri is not valid.");
+                return null;
+            }
+
+            GltfObject gltfObject;
+            bool isGlb = false;
+            bool loadAsynchronously = Application.isPlaying;
+
+            if (loadAsynchronously) { await BackgroundThread; }
+
+            if (uri.Contains(".gltf"))
```

Comment:
may want to make sure that this .gltf extension and the below .glb extension are at the end of the file. I'm not sure if youd ever get handed a .gltf.meta file, but you could have a more robust check here

Reply:
.EndsWith might be better

Reply:
yeah that sounds great

Reply:
picking up this fix